src/web/app/seed_loader.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { variations as proto } from '../../proto/generated/proto_bundle';
+import { SeedType, type ProcessingOptions } from '../../core/base_types';
+import { StudyListModel, StudyModel } from './study_model';
+import { ProcessedStudy } from '../../core/study_processor';
+
+import * as url_utils from '../../core/url_utils';
+
+async function loadFile(
+  url: string,
+  responseType: 'arraybuffer' | 'text',
+): Promise<any> {
+  return await new Promise<any | undefined>((resolve, reject) => {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', url, true /* async */);
+    xhr.responseType = responseType;
+    xhr.onload = () => {
+      if (xhr.status === 200) {
+        resolve(xhr.response);
+      } else {
+        reject(new Error('HTTP status:' + xhr.status));
+      }
+    };
+    xhr.onerror = (err) => {
+      reject(err);
+    };
+    xhr.send(null);
+  });
+}
+
+const getCurrentMajorVersion = new Promise<number>((resolve) => {
+  loadFile(url_utils.getUsedChromiumVersionUrl, 'text')
+    .then((chromeVersionData) => {
+      if (chromeVersionData !== undefined)
+        resolve(chromeVersionData.split('.')[0] ?? 0);
+      resolve(0);
+    })
+    .catch(console.error);
```

Comment:
did you want this catch to be above the `.then` so that you always resolve `0` if there's an error? Alternatively, call `resolve(0)` inside `catch` function

Reply:
Used `resolved(0)` in `catch()`